src/ExpressionEvaluator/Core/Source/ResultProvider/ResultProvider.cs

```diff
@@ -901,11 +1005,12 @@ internal void Execute()
                     {
                         completionRoutine();
                     }
-                    catch (Exception e)
+                    catch (Exception e) when (ExpressionEvaluatorFatalError.ReportNonFatalException(e, DkmComponentManager.ReportCurrentNonFatalException))
```

Comment:
I don't think we want an Exception filter here.  I think we only want the filter in cases where we can't reasonably continue (cases where we'd pass an Exception to the caller).  In this case we're saying it's okay to catch the Exception and continue by calling ```_onException``` (returning a well-formed error results), so I think we should just do that.

We could probably delete ```ExpressionEvaluatorFatalError.ReportNonFatalException```, as I don't see it called elsewhere.